applications/tests/test_applications_mutations.py

```diff
@@ -702,3 +717,315 @@ def test_delete_winter_storage_application_inexistent_application(
+            lastName
+            phoneNumber
+            email
+            address
+            zipCode
+            municipality
+            boatWidth
+            boatLength
+            informationAccuracyConfirmed
+            acceptFitnessNews
+            acceptLibraryNews
+            acceptOtherCultureNews
+            acceptBoatingNewsletter
+            harborChoices {
+                harbor {
+                    id
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def test_update_berth_application_by_owner(
+    berth_customer_api_client, berth_application, customer_profile
+):
+    berth_application_id = to_global_id(BerthApplicationNode, berth_application.id)
+    remove_choice = HarborChoiceFactory(application=berth_application)
+
+    berth = BerthFactory()
+    harbor_node_id = to_global_id(HarborNode, berth.pier.harbor.id)
+
+    customer_profile.user = berth_customer_api_client.user
+    customer_profile.save()
+    berth_application.customer = customer_profile
+    berth_application.save()
+
+    variables = {
+        "id": berth_application_id,
+        "language": "en",
+        "firstName": "John",
+        "lastName": "Doe",
+        "phoneNumber": "1234567890",
+        "email": "john.doe@example.com",
+        "address": "Mannerheimintie 1",
+        "zipCode": "00100",
+        "municipality": "Helsinki",
+        "boatWidth": Decimal("2.00"),
+        "boatLength": Decimal("3.00"),
+        "informationAccuracyConfirmed": True,
+        "acceptFitnessNews": False,
+        "acceptLibraryNews": False,
+        "acceptOtherCultureNews": False,
+        "acceptBoatingNewsletter": True,
+        "addChoices": [{"harborId": harbor_node_id, "priority": 1}],
+        "removeChoices": [to_global_id(HarborChoiceType, remove_choice.id)],
+    }
+
+    assert berth_application.harborchoice_set.count() == 1
+
+    executed = berth_customer_api_client.execute(
+        UPDATE_BERTH_APPLICATION_OWNER_MUTATION, input=variables
+    )
+
+    assert executed == {
+        "data": {
+            "updateBerthApplication": {
+                "berthApplication": {
+                    "id": berth_application_id,
+                    "language": "EN",
+                    "firstName": "John",
+                    "lastName": "Doe",
+                    "phoneNumber": "1234567890",
+                    "email": "john.doe@example.com",
+                    "address": "Mannerheimintie 1",
+                    "zipCode": "00100",
+                    "municipality": "Helsinki",
+                    "boatWidth": "2.00",
+                    "boatLength": "3.00",
+                    "informationAccuracyConfirmed": True,
+                    "acceptFitnessNews": False,
+                    "acceptLibraryNews": False,
+                    "acceptOtherCultureNews": False,
+                    "acceptBoatingNewsletter": True,
+                    "harborChoices": [{"harbor": {"id": harbor_node_id}}],
+                }
```

Comment:
Trivial thing, but this could be a merge of the "variables" -variable and the changed entries so it would be easy to see which values are changed https://stackoverflow.com/a/26853961/784642

Reply:
Hmmm that's a good idea, I think this one is a bit more explicit, and since we're already doing the assertions like this we'll keep it for consistency 😁